src/classes/services/WeatherService.cls

```diff
@@ -41,18 +41,21 @@ public with sharing class WeatherService {
 
     class PatchWeather implements WeatherRestCall {
         public String handle(RestRequest request) {
+            System.debug(request.requestBody.toString());
             WeatherDto weatherDto = (WeatherDto) JSON.deserialize(request.requestBody.toString(), WeatherDto.class);
             Weather__c weather = weatherDto.updateWeather();
             upsert weather CityId__c;
             return JSON.serialize(weather);
-
         }
     }
 
     class DeleteWeather implements WeatherRestCall {
         public String handle(RestRequest request) {
             String weatherId = request.params.get('id');
             List<Weather__c> weatherToDelete = WeatherSelector.getWeatherByCityId(weatherId);
+            if (weatherToDelete.size() == 0) {
+                return JSON.serialize('Wrong id');
```

Comment:
MINOR
But if the list is empty that means there is no data to return by Id, but not wrong Id.
Better message: "No data found."
Also, you can check `weatherToDelete.isEmpty()` which is OOTB method